### PR TITLE
[libassert] Add 2.0.0

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libassert/portfile.cmake [hunk 1]
```diff
@@ -1,37 +1,31 @@
...
     HEAD_REF main
-    PATCHES
-        runtime_destination.patch
-        magic-enum.patch
-        include-dir.patch
-        cpptrace-0.4.1.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
+set(BUILD_SHARED NOT BUILD_STATIC)
```
Comment (dg0yt) [inline ports/libassert/portfile.cmake:10]:
```suggestion
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" BUILD_SHARED)
```

File: ports/libassert/portfile.cmake [hunk 2]
```diff
@@ -1,37 +1,31 @@
...
-        -DASSERT_USE_EXTERNAL_CPPTRACE=ON
-        -DASSERT_STATIC=${BUILD_STATIC}
+      -DLIBASSERT_USE_EXTERNAL_CPPTRACE=ON
+      -DLIBASSERT_USE_EXTERNAL_MAGIC_ENUM=ON
+      -DLIBASSERT_BUILD_SHARED=${BUILD_SHARED}
 )
 
 vcpkg_cmake_install()
 vcpkg_cmake_config_fixup(
     PACKAGE_NAME assert
```
Comment (dg0yt) [inline ports/libassert/portfile.cmake:22]: If the config is now stored in libassert, it is probably no longer named "assert". But I didn't check.
```suggestion
```

File: ports/libassert/portfile.cmake [hunk 3]
```diff
@@ -1,37 +1,31 @@
...
-    CONFIG_PATH lib/cmake/assert
+    CONFIG_PATH lib/cmake/libassert
 )
 vcpkg_copy_pdbs()
 
-file(APPEND "${CURRENT_PACKAGES_DIR}/share/assert/assert-config.cmake" "include(CMakeFindDependencyMacro)
-find_dependency(magic_enum)
-find_dependency(cpptrace)")
+file(APPEND "${CURRENT_PACKAGES_DIR}/share/libassert/libassert-config.cmake" "find_dependency(magic_enum)")
```
Comment (dg0yt) [inline ports/libassert/portfile.cmake:27]: So package name is "libassert" now, indeed.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/libassert/portfile.cmake (hunk 2 above)
Comment (jeremy-rifkin) [reply ports/libassert/portfile.cmake:22]: Thanks!

File: ports/libassert/portfile.cmake [hunk 4]
```diff
@@ -1,37 +1,30 @@
...
     HEAD_REF main
-    PATCHES
-        runtime_destination.patch
-        magic-enum.patch
-        include-dir.patch
-        cpptrace-0.4.1.patch
 )
 
-string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dyanmic" BUILD_SHARED)
```
Comment (dg0yt) [inline ports/libassert/portfile.cmake:9]:
```suggestion
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" BUILD_SHARED)
```
Typo. Sorry :facepalm:

Comment (jeremy-rifkin) [reply ports/libassert/portfile.cmake:9]: Oops thanks

File: ports/libassert/portfile.cmake [hunk 5]
```diff
@@ -1,37 +1,30 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO jeremy-rifkin/libassert
-    REF "v${VERSION}"
-    SHA512 beba94e033f7e43c84123736a32725a333c915392d5dc57c26a63f832a507564d79f290a151cb136de8bded3d8d343dad3c4bf2efec9977d878df3c9a8677554
+    REF "v${VERSION}-beta"
```
Comment (jeremy-rifkin) [inline ports/libassert/portfile.cmake:4]: temporary while I sort out package manager stuff, then I'll do the v2.0.0 release upstream

**Review by Cheney-W — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**